src/gpgpu-sim/shader.h

```diff
@@ -1314,6 +1355,13 @@ class ldst_unit : public pipelined_simd_unit {
             const memory_config *mem_config, class shader_core_stats *stats,
             unsigned sid, unsigned tpc);
 
+  // Ni: Add a structure to record the LDGSTS instructions,
+  // similar to m_pending_writes, but since LDGSTS does not have a output register
+  // to write to, so a new structure needs to be added
+  std::map<unsigned /*warp_id*/,
+           std::map<unsigned /*pc*/, 
+                  std::map<unsigned /*addr*/, unsigned /*count*/>>>
+      m_pending_ldgsts;
```

Comment:
Nice documentation for the meaning of each level of key. However you can further
improve the comments to make it compatible with common documentation tools, such
that the meanings of each key can show up in VS Code intellisense (hover your
mouse over the variable name). Try this:

```suggestion
  /**
   * A multi-level map: unsigned (warp_id) -> unsigned (pc) -> unsigned (addr) -> unsigned (count)
   */
  std::map<unsigned /*warp_id*/,
           std::map<unsigned /*pc*/, 
                  std::map<unsigned /*addr*/, unsigned /*count*/>>>
      m_pending_ldgsts;
```